### PR TITLE
Fix for Add, Update & Delete of Objective & Funding Source

### PR DESCRIPTION
**Pull request by ailZhou**
### Description

There is an issue currently in the data structure for add/edit/delete of cards in the initiative dashboard section for Objective & Funding Source due to the nesting of the forms.

OLD: Only 1 objective was able to be saved because the data was saving at the initiative level
old

NEW: There is now an array stored inside initiatives using entityType as the object key
new

NOTE: ignore the getEntityStatus.tsx changes, they will be use for the entity status later. This code snuck in but won't cause any problems if merged in now

Will be addressed in another PR:
- There is an issue with saved nested data not being render in the form when you try to edit the form after saving. If you reload the initiative dashboard, it will show up. You can see it in Objective when adding quantitive targets.
- Save & Return button not doing anything
- Code coverage will be increased later in separate ticket

### Related ticket(s)

CMDCT-193

---
### How to test

1) Sign into MFP
2) Create a new WP
3) Inside the new WP, go to "State and Territory-Specific Initiatives"
4) Add Initiative
5) Click "Edit" on the newly added Initiative 
6) In Evaluation Plan or Funding Sources, test adding, editing and deleting card. There should not be any issues when doing those actions and multiple of each should be able to coexist

### Important updates

N/A

---
### Author checklist

- [x] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/modals/AddEditOverlayEntityModal.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+} from "types";
+import {
+  entityWasUpdated,
+  filterFormData,
+  getEntriesToClear,
+  setClearedEntriesToDefaultValue,
+  useStore,
+} from "utils";
+
+export const AddEditOverlayEntityModal = ({
```
[the remark this answers is not in the document]
Comment (ailZhou) [reply services/ui-src/src/components/modals/AddEditOverlayEntityModal.tsx:23]: i don't have to listen to you!

**Review by ajaitasaini — COMMENTED** [review]
https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/6752256/cf7714e6-f9ec-4b6d-abfa-97bcd9a75597

i'm seeing way too many repeated fields under Funding source (it's duplicating far more than it needs to and also in the wrong order?). i wanted to confirm whether it is related to your changes or if this requires a separate bug ticket!

**Review by cassandradanger — APPROVED** [review]
I'm not seeing the same issue that Ajaita is seeing...everything looks good in the browser for me! But I can re-review if updates are needed

**Comment by ailZhou** [comment]
> Member

i tried to trigger it both locally and on deploy, i'm not seeing the issue. it could just be an issue of dirty data? can you see if a new form is causing that?

**Review by keeysnc — COMMENTED** [review]
For funding sources, if you add text in the Other specify its not showing up in the card when saving. It is however being saved because when I open up the modal its still showing the text. 
Screenshot 2023-10-17 at 12 21 15 PM

In entities.ts within the getRadioValue method, the property initiative_wp_otherTopic is being used to get the "Specify Other" text but for funding sources that property is fundingSources_wp_otherTopic. I think it just needs to be updated in the wp.json.

**Review by gmrabian — APPROVED** [review]
Verified locally – thanks for the walkthrough!

**Review by karla-vm — APPROVED** [review]

**Pull request merged**